packages/trip-form/src/MetroModeSelector/utils.ts

```diff
@@ -190,3 +190,64 @@ export const setModeButtonEnabled = (enabledKeys: string[]) => (
     enabled: enabledKeys.includes(modeButton.key)
   };
 };
+
+/**
+ * Determines which modeButton (and potentially modeSetting) needs to be set in the query params
+ * for a certain TransportMode to be enabled. This is necessary because certain configurations
+ * have combinations of modeButtons and modeSettings that do not directly map to a given mode
+ * @param modeButtons Array of mode buttons from config
+ * @param settings Array of mode settings from config
+ * @param transportMode Desired transport mode
+ * @returns The relevant modeButton and modeSetting, or undefined if no combination is found
+ */
+export const findRequiredOptionsForTransportMode = (
+  modeButtons: ModeButtonDefinition[],
+  settings: ModeSetting[],
+  transportMode: TransportMode
+):
+  | {
+      modeSetting?: string;
+      modeButton: string;
+    }
+  | undefined => {
```

Comment:
I'll leave this up to you to decide: what do you think is easier to handle as a consumer of this function:
- making both modeSetting and modeButton optional
- as you have now, making the entire object optional

I was thinking 1 would be cleaner here but maybe there are advantages to the consumer for the way you've done it